### PR TITLE
Update code and tests to still build and test with Java 7

### PR DESCRIPTION
**Pull request by jhanders34**
- Update rest.handler.config bundle to be Java 7 byte code to make restConnector-2.0 feature work with Java 7
- Change tests to use Java EE 7 features instead of EE8 features and add RepeatTests to test them with Java EE 8 features as well.
- Update Java EE8 or MP tests that require Java 8, to have a MinimumJavaLevel set to Java 8 so they don't run with Java 7
- Update logic to get caller to be generic and work on different Java levels due to number of stack elements in the thread stack being different.
- Update transaction.test.util to be Java 7 to support com.ibm.ws.wsat tests running with Java 7.

### CODE REVIEW

**Comment by jhanders34** [comment]
I have added several people on review for this PR:

Paul to review the http compression test changes
Kyle to review the jca test changes
Jon to review the wsat related test changes
Andy to review the JAXRS test changes
Don to review the messaging test changes
Ben to review the cdi test changes

Of course please review any of the other changes that are not part of the changes I want you to focus on.

**Comment by jhanders34** [comment]
I have added several people on review for this PR:

Paul to review the http compression test changes
Kyle to review the jca test changes
Jon to review the wsat related test changes
Andy to review the JAXRS test changes
Don to review the messaging test changes
Ben to review the cdi test changes

Of course please review any of the other changes that are not part of the changes I want you to focus on.

**Comment by jhanders34** [comment]
#build

**Comment by jhanders34** [comment]
I have also done a full fat build with this change for both Java 7 and Java 8 to verify that things are working in both for the tests and code changed.

File: dev/com.ibm.ws.jca_fat/bnd.bnd [hunk 1]
```diff
@@ -18,6 +18,11 @@ src: \
 
 fat.project: true
 
+# Define additional tested features that are NOT present any any XML files in this bucket.
```
Comment (KyleAure) [inline dev/com.ibm.ws.jca_fat/bnd.bnd]: Minor Comment: should read "in any" ... I think the same typo got copied to the other bnd files as well.

Comment (jhanders34) [reply dev/com.ibm.ws.jca_fat/bnd.bnd]: Will fix.  What is bad is I copied it from another place as well.

Comment (jhanders34) [reply dev/com.ibm.ws.jca_fat/bnd.bnd]: I have made the changes.  If you are satisfied please mark the conversation resolved.

**Comment by jhanders34** [comment]
#run-libby-bot

**Pull request merged**